volatility3/framework/plugins/windows/windows.py

```diff
@@ -0,0 +1,137 @@
+                version=(1, 0, 0),
+            ),
+        ]
+
+    @classmethod
+    def list_windows(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel_module_name: str,
+    ) -> Iterable[interfaces.objects.ObjectInterface]:
+        """
+        Enumerates the desktops of each window station
+        For each found, enumerates its windows within the desktop
+        """
+        kernel = context.modules[kernel_module_name]
+
+        for (
+            winsta,
+            station_name,
+            session_id,
+        ) in windowstations.WindowStations.scan_window_stations(
+            context, config_path, kernel_module_name
+        ):
+            # for each window station, walk its list of desktops
+            for desktop, desktop_name in winsta.desktops(kernel.symbol_table_name):
+                try:
+                    top_window = desktop.pDeskInfo.spwnd
+                except exceptions.InvalidAddressException:
+                    vollog.debug(
+                        f"Desktop with name {desktop_name} in window station {station_name} has a broken window pointer."
+                    )
+                    continue
+
+                for window, window_name in desktop.windows(top_window):
+                    yield station_name, desktop_name, window, window_name
+
+    def _generator(self):
+        kernel_name = self.config["kernel"]
+
+        # call the implementation for finding windows and gather attributes
+        for station_name, desktop_name, window, window_name in self.list_windows(
+            self.context, self.config_path, kernel_name
+        ):
+            # We need a valid process and session id for the window to display it
+            process = window.get_process()
+            process_name = None
+            if process:
+                try:
+                    process_name = utility.array_to_string(process.ImageFileName)
+                    process_pid = process.UniqueProcessId
+                except exceptions.InvalidAddressException:
+                    vollog.debug(
+                        f"Unable to read name and pid of the process for window {window.vol.offset:#x}"
+                    )
+
+            if process_name is None:
+                vollog.warning(
+                    f"Invalid process reference for the process hosting window {window.vol.offset:#x}"
+                )
+                continue
+
+            sess_id = window.get_session_id()
+            if sess_id is None:
+                vollog.debug(
+                    f"Unable to read session id of the process for window {window.vol.offset:#x} in process {process_name}"
+                )
+                continue
+
+            # procedures can be empty, but if set, should be a valid pointer
+            window_proc = window.get_window_procedure()
+            if window_proc is None or window_proc == 0 or window_proc > 0x1000:
```

Comment:
Think this logic is the wrong way around?  Currently says if window_proc is None windows_proc = format_hints.Hex(None)...  5:S

Reply:
Changed it to the NotAvailable on None. I thought there was a recent update where format_hints returned NotAvailable when None was sent in. I guess I misunderstood that PR.